src/components/track-tile/TrackTile.tsx

```diff
@@ -0,0 +1,190 @@
+import { flexCol, flexRow } from 'app/styles'
+import { ThemeColors } from 'app/utils/theme'
+
+import TrackBannerIcon, { TrackBannerIconType } from './TrackBannerIcon'
+import TrackTileBottomButtons from './TrackTileBottomButtons'
+import TrackTileCoSign from './TrackTileCoSign'
+import TrackTileMetadata from './TrackTileMetadata'
+import TrackTileStats from './TrackTileStats'
+import TrackTileTopRight from './TrackTileTopRight'
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    container: {
+      ...flexRow(),
+      minHeight: 152,
+      borderColor: themeColors.neutralLight8,
+      backgroundColor: themeColors.white,
+      borderWidth: 1,
+      borderRadius: 8,
+      maxWidth: 400,
+      marginHorizontal: 'auto',
+      marginBottom: 12,
+      elevation: 3,
+      shadowOpacity: 0.15,
+      shadowOffset: { width: 0, height: 1 },
+      shadowRadius: 3
+    },
+    mainContent: {
+      ...flexCol(),
+      flex: 1
+    }
+  })
+
+const TrackTile = ({
+  artistName,
+  coSign,
+  coverArtSizes,
+  duration,
+  fieldVisibility,
+  goToTrackPage,
+  goToArtistPage,
+  hasCurrentUserReposted,
+  hasCurrentUserSaved,
+  id,
+  index,
+  isArtistPick,
+  isOwner,
+  isPlaying,
+  isTrending,
+  isUnlisted,
+  listenCount,
+  makeGoToFavoritesPage,
+  makeGoToRepostsPage,
+  onClickOverflow,
+  onLoad,
+  onShare,
+  repostCount,
+  saveCount,
+  showArtistPick,
+  showRankIcon,
+  showSkeleton,
+  title,
+  togglePlay,
+  toggleRepost,
+  toggleSave,
+  user,
+  uid
+}: TrackTileProps) => {
+  const opacity = useRef(new Animated.Value(0)).current
+  const fadeIn = { opacity }
+
+  const styles = useThemedStyles(createStyles)
+  const hideShare: boolean = fieldVisibility
+    ? fieldVisibility.share === false
+    : false
+  const hidePlays = fieldVisibility
+    ? fieldVisibility.play_count === false
+    : false
+
+  const onToggleSave = useCallback(() => toggleSave(id), [toggleSave, id])
+
+  const onToggleRepost = useCallback(() => toggleRepost(id), [toggleRepost, id])
+
+  const onClickShare = useCallback(() => onShare(id), [onShare, id])
+
+  const onClickOverflowMenu = useCallback(
+    () => onClickOverflow && onClickOverflow(id),
+    [onClickOverflow, id]
+  )
+
+  const [artworkLoaded, setArtworkLoaded] = useState(false)
+
+  const isLoaded = artworkLoaded && !showSkeleton
+
+  useEffect(() => {
+    if (isLoaded) {
+      onLoad(index)
+      Animated.timing(opacity, {
+        toValue: 1,
+        easing: Easing.ease,
+        useNativeDriver: true
+      }).start()
+    }
+  }, [onLoad, isLoaded, index, opacity])
+
+  return (
+    <View style={styles.container}>
+      {showArtistPick && isArtistPick && (
+        <TrackBannerIcon type={TrackBannerIconType.STAR} />
+      )}
+      {isUnlisted && <TrackBannerIcon type={TrackBannerIconType.HIDDEN} />}
+      <Pressable
+        style={styles.mainContent}
+        onPress={() => {
+          if (showSkeleton) return
```

Comment:
is there a way to use `disabled` prop for pressable, so we don't have to complicate the onPress function?

Reply:
There is indeed, will update!